### PR TITLE
refactor(scheduler): join frequency and count in `RetryMapping`

### PR DESCRIPTION
**Pull request by Sakilmostak**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

frequency and count are different field in RetryMapping that are conjoined to simplify the logic to set schedule time

## How did you test it?

- Run a scheduler
- Run a consumer
- Create a MCA(Cybersource)
- Create a Card Non-3DS payment
- Status should be succeeded
- Do a DB query to check whether process tracker is created for Payment Sync 
- PSync tracker must be present against the transaction made

Note: This affect scheduling of Process Trackers

Testcases added in the linked issue
## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/scheduler/src/utils.rs [hunk 1]
```diff
@@ -356,28 +350,23 @@ pub fn get_outgoing_webhook_retry_schedule_time(
...
-                .iter()
-                .zip(retry_mapping.frequency.iter()),
-        )
+        get_delay(retry_count, &retry_mapping.frequencies)
     }
 }
 
 /// Get the delay based on the retry count
-fn get_delay<'a>(retry_count: i32, array: impl Iterator<Item = (&'a i32, &'a i32)>) -> Option<i32> {
+fn get_delay(retry_count: i32, vec: &[(i32, i32)]) -> Option<i32> {
```
Comment (SanchithHegde) [inline crates/scheduler/src/utils.rs:358]:
```suggestion
fn get_delay(retry_count: i32, frequencies: &[(i32, i32)]) -> Option<i32> {
```

File: crates/scheduler/src/utils.rs [hunk 2]
```diff
@@ -356,28 +350,23 @@ pub fn get_outgoing_webhook_retry_schedule_time(
...
     let mut cumulative_count = 0;
-    for (&count, &frequency) in array {
+    let itervec = vec.iter();
+    for (frequency, count) in itervec {
         cumulative_count += count;
         if cumulative_count >= retry_count {
-            return Some(frequency);
+            return Some(frequency.to_owned());
         }
     }
```
Comment (SanchithHegde) [inline crates/scheduler/src/utils.rs:371]:
```suggestion
    for (&frequency, &count) in frequencies {
        cumulative_count += count;
        if cumulative_count >= retry_count {
            return Some(frequency);
        }
    }
```

File: crates/scheduler/src/utils.rs [hunk 3]
```diff
@@ -356,28 +350,23 @@ pub fn get_outgoing_webhook_retry_schedule_time(
...
-                .zip(retry_mapping.frequency.iter()),
-        )
+        get_delay(retry_count, &retry_mapping.frequencies)
     }
 }
 
 /// Get the delay based on the retry count
-fn get_delay<'a>(retry_count: i32, array: impl Iterator<Item = (&'a i32, &'a i32)>) -> Option<i32> {
+fn get_delay(retry_count: i32, vec: &[(i32, i32)]) -> Option<i32> {
     // Preferably, fix this by using unsigned ints
```
Comment (SanchithHegde) [inline crates/scheduler/src/utils.rs:359]: Well, this is another thing that we should address. Can be in a separate PR, not a problem.

File: crates/scheduler/src/utils.rs [hunk 4]
```diff
@@ -356,28 +350,22 @@ pub fn get_outgoing_webhook_retry_schedule_time(
...
         return None;
     }
 
     let mut cumulative_count = 0;
-    for (&count, &frequency) in array {
+    for (frequency, count) in frequencies {
         cumulative_count += count;
         if cumulative_count >= retry_count {
-            return Some(frequency);
+            return Some(frequency.to_owned());
```
Comment (SanchithHegde) [inline crates/scheduler/src/utils.rs:368]: Do we need the explicit `to_owned()`?

**Review by SanchithHegde — APPROVED** [review]

**Review by Narayanbhat166 — APPROVED** [review]
